cuda/matrix/csr_kernels.cu

```diff
@@ -303,59 +303,59 @@ void spmv(std::shared_ptr<const CudaExecutor> exec,
             syn::value_list<int>(), syn::type_list<>(), exec, a, b, c);
     } else if (a->get_strategy()->get_name() == "sparselib" ||
                a->get_strategy()->get_name() == "cusparse") {
-        if (cusparse::is_supported<ValueType, IndexType>::value) {
+        auto handle = exec->get_cusparse_handle();
+        {
+            cusparse::pointer_mode_guard pm_guard(handle);
+            const auto alpha = one<ValueType>();
+            const auto beta = zero<ValueType>();
             // TODO: add implementation for int64 and multiple RHS
-            auto handle = exec->get_cusparse_handle();
-            {
-                cusparse::pointer_mode_guard pm_guard(handle);
-                const auto alpha = one<ValueType>();
-                const auto beta = zero<ValueType>();
-                // TODO: add implementation for int64 and multiple RHS
-                if (b->get_stride() != 1 || c->get_stride() != 1)
-                    GKO_NOT_IMPLEMENTED;
+            if (b->get_stride() != 1 || c->get_stride() != 1)
+                GKO_NOT_IMPLEMENTED;
 
 #if defined(CUDA_VERSION) && (CUDA_VERSION < 11000)
-                auto descr = cusparse::create_mat_descr();
-                auto row_ptrs = a->get_const_row_ptrs();
-                auto col_idxs = a->get_const_col_idxs();
-                cusparse::spmv(handle, CUSPARSE_OPERATION_NON_TRANSPOSE,
-                               a->get_size()[0], a->get_size()[1],
-                               a->get_num_stored_elements(), &alpha, descr,
-                               a->get_const_values(), row_ptrs, col_idxs,
-                               b->get_const_values(), &beta, c->get_values());
-
-                cusparse::destroy(descr);
+            // TODO: add implementation for int64 and multiple RHS
+            if (!cusparse::is_supported<ValueType, IndexType>::value) {
+                GKO_NOT_IMPLEMENTED;
+            }
+
+            auto descr = cusparse::create_mat_descr();
```

Comment:
Could you add the structure to handle the create/destroy like cublas/cusparse handle?
also dn_vec and sp_mat.
I did not add it properly when I push it :sweat_smile:

Reply:
Maybe we could make that part of another PR? I already started updating the SpMV in #905, maybe I can include that there? This is a rather small fix. What do you mean more precisely? RAII for handles?

Reply:
I see, yes RAII for these mat_descr